packages/dd-trace/test/appsec/passport.spec.js

```diff
@@ -0,0 +1,239 @@
+        rootSpan,
+        'safe'
+      )
+    })
+
+    it('should report login success and the extended fields in extended mode', () => {
+      const user = userUuid
+
+      user.id = 'publicName'
+      passportModule.passportTrackEvent(loginLocal, user, rootSpan, 'extended')
+      expect(setUser.setUserTags).to.have.been.calledOnceWithExactly(
+        {
+          id: user.id,
+          login: 'test',
+          email: 'testUser@test.com',
+          username: 'Test User'
+        },
+        rootSpan
+      )
+      expect(events.trackEvent).to.have.been.calledOnceWithExactly(
+        'users.login.success',
+        {},
+        'passportTrackEvent',
+        rootSpan,
+        'extended'
+      )
+    })
+
+    it('should not call trackEvent in safe mode if sdk user event functions are already called', () => {
+      rootSpan.context = () => {
+        return {
+          _tags: {
+            '_dd.appsec.events.users.login.success.sdk': 'true'
+          }
+        }
+      }
+
+      passportModule.passportTrackEvent(loginLocal, userUuid, rootSpan, 'safe')
+      expect(setUser.setUserTags).not.to.have.been.called
+      expect(events.trackEvent).not.to.have.been.called
+    })
+
+    it('should not call trackEvent in extended mode if sdk user event functions are already called', () => {
+      rootSpan.context = () => {
+        return {
+          _tags: {
+            '_dd.appsec.events.users.login.success.sdk': 'true'
+          }
+        }
+      }
+
+      passportModule.passportTrackEvent(loginLocal, userUuid, rootSpan, 'extended')
+      expect(setUser.setUserTags).not.to.have.been.called
+      expect(events.trackEvent).not.to.have.been.called
+    })
+
+    it('should call trackEvent in extended mode if sdk custom event function is already called', () => {
+      rootSpan.context = () => {
+        return {
+          _tags: {
+            '_dd.appsec.events.custom.event.sdk': 'true'
+          }
+        }
+      }
+
+      passportModule.passportTrackEvent(loginLocal, userUuid, rootSpan, 'extended')
+      expect(events.trackEvent).to.have.been.calledOnceWithExactly(
+        'users.login.success',
+        {},
+        'passportTrackEvent',
+        rootSpan,
+        'extended'
+      )
+    })
+
+    it('should call trackEvent in extended mode if sdk dy called', () => {
+      rootSpan.context = () => {
+        return {
+          _tags: {
+            '_dd.appsec.events.users.login.success.sdk': 'true'
+          }
+        }
+      }
+
+      passportModule.passportTrackEvent(loginLocal, userUuid, rootSpan, 'extended')
+      expect(setUser.setUserTags).not.to.have.been.called
+      expect(events.trackEvent).not.to.have.been.called
+    })
+
+    it('should report login success with the _id field', () => {
+      const user = {
+        _id: '591dc126-8431-4d0f-9509-b23318d3dce4',
+        email: 'testUser@test.com',
+        username: 'Test User'
+      }
+
+      rootSpan.context = () => { return {} }
```

Comment:
what you're doing here is cleaning up a side effect from a previous test. You shouldn't be doing that here, instead you should be resetting this value in the beforeEach at the start of the file